types/coin.go

```diff
@@ -172,6 +172,30 @@ func (coins Coins) IsValid() bool {
 	}
 }
 
+// AddCoinByDenom looks for a matching coin in coins that has the same denom as
+// other. Upon matching, the other coin is added to the matching coin. If the
+// coins are empty, the coin is simply added to the empty set. If the denom of
+// the other coin to be added does not exist in coins, the function returns the
+// original coins.
+func (coins Coins) AddCoinByDenom(other Coin) Coins {
```

Comment:
So this function will return `Coins{Coin}` if `Coins` is empty, but if `Coins` doesn't contain any of `Coin.denom` then it doesn't return with `Coin` included in `Coins`. That seems like it violates expectations. Can we make it so `Coins` includes `Coin` if the denom isn't present in `Coins`? cc @alexanderbez

Reply:
I'll revise it to not add `coin` when the `coins` set is empty.